src/modules/companies-branches/index.ts

```diff
@@ -0,0 +1,98 @@
+import { ClientType } from 'index';
+
+import {
+  CREATE_COMPANY_BRANCH,
+  UPDATE_COMPANY_BRANCH,
+  DELETE_COMPANY_BRANCH,
+  ADD_USER_TO_BRANCH,
+  REMOVE_USER_FROM_BRANCH,
+} from '../../mutations';
+import {
+  UserInterface,
+  CompanyBranchInput,
+  CompanyBranchInterface,
+  QueryBranchesWhereWhereConditions,
+  QueryBranchesOrderByOrderByClause,
+  QueryCompanyBranchUsersWhereWhereConditions,
+} from '../../types';
+import {
+  COMPANIES_BRANCHES_QUERY,
+  COMPANIES_BRANCHES_USER_QUERY,
+} from '../../queries';
+
+export class CompaniesBranches {
+  constructor(protected client: ClientType) {}
+
+  public async getCompanyBranches(
+    where: QueryBranchesWhereWhereConditions,
+    orderBy: QueryBranchesOrderByOrderByClause,
+    first?: number,
+    page?: number
+  ): Promise<CompanyBranchInterface> {
+    const response = await this.client.query({
+      query: COMPANIES_BRANCHES_QUERY,
+      variables: { where, orderBy, first, page },
+    });
+    return response.data as CompanyBranchInterface;
+  }
+
+  public async getCompanyBranchUsers(
+    where: QueryCompanyBranchUsersWhereWhereConditions,
+    first?: number,
+    page?: number
+  ): Promise<UserInterface> {
+    const response = await this.client.query({
+      query: COMPANIES_BRANCHES_USER_QUERY,
+      variables: { where, first, page },
+    });
+    return response.data as UserInterface;
+  }
+
+  public async createCompanyBranch(
+    input: CompanyBranchInput
+  ): Promise<CompanyBranchInterface> {
+    const response = await this.client.mutate({
+      mutation: CREATE_COMPANY_BRANCH,
+      variables: { input },
+    });
+    return response.data.createCompanyBranch;
```

Comment:
**question(non-blocking):** I see this function's return is not being cast as the expected return type. Is the casting not needed? Did you test to see if this works as intended?